### ISSUE TITLE
Focus out after adding an image doesn't update the model which causes the image to be deleted - Firefox 35.0.1

### ISSUE THREAD

**Issue by GuyGolanBB** [fraywing/textAngular#528]

**Comment by SimeonC** [comment]
OK, there's something really bad going on with images in FF, it seems that adding an image in FF is deleting the `<p>` tag that should be wrapped around it which helps with formatting and such. But it seems that the image is just flat out replacing the whole lot and doing other weird stuff afterwards.

I'll look into it.

**Comment by GuyGolanBB** [comment]
Thanks

**Comment by oscar-g** [comment]
This is pretty strange. Inserting an image seems to work when the model is not empty (for exmaple, if there is an empty <p></p>)...if there is nothing in the editor, I am able to insert the image, but it disappears when I remove focus from the editor.

**Comment by oosidat** [comment]
I'm not sure if this is entirely related, but here's what I found in my investigation - hope it's helpful to someone debugging this issue.

* A problem occurs in Firefox when `insertImage` is used on an empty editor
* Placing a `$scope.$watch(...)` along with some logging in `demo/textAngular.comhtml` indicates that an `insertImage` action does not change the value of `ng-model` (in this case, `htmlContent`) in Firefox, but does so in Chrome - despite the image being displayed.
* I traced the code back to line 41 of `src/DOM.js` (`selectedElement = taSelection.getSelectionElement();`). Console logging `selectedElement`, and observing its value on the `insertImage` action reveals two entirely different values in Firefox and Chrome.

**Comment by oscar-g** [comment]
Thanks, @oosidat, this confirms what I observed.Do you mind posting the the value of ```selectedElement``` on Chrome vs FF?

**Comment by oscar-g** [comment]
I just noticed this also happens in IE 11

**Comment by oosidat** [comment]
##### Steps:

* Open `demo/textangular.com.html`
* Click `insertImage`
* Add image link to dialog
* Click `Ok`

##### Chrome:
```
<h2>
    <img src="http://www.newstalk1010.com/pics/Feeds/Articles/2015723/1319515/Cat.jpg">
    <br>
</h2>
```

##### Firefox:
```
<div contenteditable="true" id="taTextElement1682513650294872" ta-bind="ta-bind" ng-model="html" class="ng-untouched ng-valid ta-bind ng-dirty ng-valid-parse">
```

I also forked the project and added a Protractor test for my fork:

https://github.com/oosidat/textAngular/commit/f31ef3bfdd364a1c8470df8e362de99b897feb01

(there's a readme in the `test/protractor` folder for running the Protractor tests)

**Comment by oscar-g** [comment]
Very weird...A single space after the image is enough to avoid this problem

**Comment by oosidat** [comment]
I think anything prefixing or suffixing the image will fix the problem - but that feels like a hack/workaround rather than a solution.

**Comment by pbassut** [comment]
Weird. This issue needs further investigation to see what the size of the problem is.
The information provided so far will get us a step ahead. Thanks guys.

**Comment by JoelParke** [comment]
<p>With the fix for #862, this may be fixed.  Can you retest with d5a508e3 and let us know???  THANKS</p>

**Comment by JoelParke** [comment]
<p>More testing and cannot reproduce with the current changes... for closing... please retest with the new release when it appears and RE-open if it is still an issue.</p>
